medusa/process_tv.py

```diff
@@ -271,7 +271,10 @@ def processDir(dirName, nzbName=None, process_method=None, force=False, is_prior
 
                 videoInRar = [x for x in rarContent if helpers.is_media_file(x)] if rarContent else ''
                 allowed_extensions = app.ALLOWED_EXTENSIONS.split(',')
-                notwantedFiles = [x for x in fileList if x not in videoFiles and x[-3:] not in allowed_extensions]
+                notwantedFiles = [x for x in fileList
+                                  if x not in videoFiles and
+                                  helpers.get_extension(x) not in allowed_extensions and
+                                  x not in rarFiles]
```

Comment:
@labrys rarfiles are unwanted only in MOVE. For hardlink/copy/symlink they are unwanted. We only delete the folder in move. For the others, rarfiles should be kept untouched